.github/workflows/pr.yaml

```diff
@@ -65,3 +67,28 @@ jobs:
       arch: "amd64"
       container_image: "rapidsai/ci:latest"
       run_script: "ci/test_notebooks.sh"
+  wheel-build-cuml:
+    secrets: inherit
+    uses: rapidsai/shared-action-workflows/.github/workflows/wheels-manylinux-build.yml@upkeep/manylinux-improvements
+    with:
+      build_type: pull-request
+      package-name: cuml
+      package-dir: python
+      extra-repo: rapidsai/cumlprims_mg
+      extra-repo-sha: branch-23.02
+      extra-repo-deploy-key: CUMLPRIMS_SSH_PRIVATE_DEPLOY_KEY
+      skbuild-configure-options: "-DCUML_BUILD_WHEELS=ON -DDETECT_CONDA_ENV=OFF -DCPM_cumlprims_mg_SOURCE=/project/cumlprims_mg/"
+  wheel-tests-cuml:
+    needs: wheel-build-cuml
+    secrets: inherit
+    uses: rapidsai/shared-action-workflows/.github/workflows/wheels-manylinux-test.yml@upkeep/manylinux-improvements
+    with:
+      build_type: pull-request
+      package-name: cuml
+      # Always want to test against latest dask/distributed.
+      test-before-amd64: "pip install git+https://github.com/dask/dask.git@main git+https://github.com/dask/distributed.git@main git+https://github.com/rapidsai/dask-cuda.git@branch-23.02"
+      # On arm also need to install cupy from the specific webpage and CMake
+      # because treelite needs to be compiled (no wheels available for arm).
+      test-before-arm64: "pip install cupy-cuda11x -f https://pip.cupy.dev/aarch64 && pip install cmake && pip install git+https://github.com/dask/dask.git@main git+https://github.com/dask/distributed.git@main git+https://github.com/rapidsai/dask-cuda.git@branch-23.02"
+      test-unittest: "pytest -v ./python/cuml/tests -k 'not test_silhouette_score_batched'"
+      test-smoketest: "" #"python ci/wheel_smoke_test_cuml.py"
```

Comment:
@dantegd  do you have any recommendations on what we could use as a reasonable smoke test for cuml? Here are the current examples for [cudf](https://github.com/rapidsai/cudf/pull/12427/files#diff-f2d06f7697ad781f29c9c3322ac1f522feaba7d01d761d3fd086721d9ab872fb) and [rmm](https://github.com/rapidsai/rmm/pull/1182/files#diff-6f7e4916411f089af7bb37a031ef77d2a63b2d365a9df81d78116af7fb68c048). For context, for PR CI we will build packages on both arm and amd architectures, but to conserve GPU CI resources we will only be running the full test suite on amd and will instead only run a small smoke test on arm. The full test suite will be run for arm on nightly builds.

Reply:
iris 👁️😉

We use this for xgboost-conda and it might be appropriate for cuml as well. https://github.com/rapidsai/xgboost-conda/blob/branch-23.02/recipes/xgboost/test-py-xgboost.py

Reply:
Thanks for the tip!

Reply:
The example that @bdice pointed out works perfectly, just changing for a cuML estimator indeed!

Reply:
Working well! https://github.com/rapidsai/cuml/actions/runs/3942397625/jobs/6746289143#step:9:1966